### PR TITLE
Pass build info together with feedback

### PR DESCRIPTION
**Pull request by orestesgaolin**
- added simple BuildInfo and PlatformBuildInfo class to retrieve env variables passed through dart-define (buildNumber, buildVersion, buildCommit)
- added BuildInfoManager that retrieves the data from injected BuildInfo
- moved deviceId to BuildInfoManager
- refactored WiredashController to have separate methods for setting user and build props
- added basic introductory tests
- updated readme and changelog
- updated sample

TODO:
- parse the provided keys on the backend @flschweiger - right now these keys can be not included, but I can easily change it to be always present but null
- maybe refactor Wiredash widget to hold only WiredashController and move state's responsibilities to controller

Tested:
- [x] running the example app with both flags passed (Flutter 1.17.1)
- [x] running the example app with only one flag passed (Flutter 1.17.1)
- [x] running the example app without any flags (Flutter 1.17.1)
- [x] running the example app without any flags fails at runtime because of [this issue](https://github.com/dart-lang/sdk/issues/41168) or something similar related to `bool.hasEnvironment` (Flutter 1.16.0)
- [x] running the example app on Flutter 1.16 or lower with flags is impossible (`Exception: The value of -DDartDefines is not formatted correctly` and sdk is currently constrained to 1.17.0 and 2.8.0)

### CODE REVIEW

**Comment by orestesgaolin** [comment]
I just noticed that we can pass versionNumber through `Wiredash.of(context).setIdentifiers(appVersion: "1.4.3");`

Shall we deprecate it or leave as it is?

**Comment by Nash0x7E2** [comment]
I would say leave it, there maybe instances here users want to manually change or set custom values here.

**Comment by orestesgaolin** [comment]
Any ideas why static analysis fails? It works on my machine

**Comment by atoka93** [comment]
@orestesgaolin for me `dartfmt` deletes two spaces in the documentation in `build_info.dart` lines 2 and 5.
Maybe that's it?

**Comment by orestesgaolin** [comment]
TODO:
- Refactor current method to:
  - setUserProperties()
  - setBuildProperties()
- Add buildVersion to dart-define way
- describe everything in readme

**Comment by flschweiger** [comment]
Yeah I really like that 😃 Maybe we should also refactor the `appVersion` property to `buildVersion` as well so the naming is more consistent.

File: lib/src/common/utils/device_info.dart [hunk 1]
```diff
@@ -3,16 +3,29 @@ import 'dart:math' show Random;
 import 'dart:ui' as ui show window;
 
 import 'package:shared_preferences/shared_preferences.dart';
+import 'package:wiredash/src/common/build_info/build_info_manager.dart';
 
 class DeviceInfo {
   static const _prefsDeviceID = '_wiredashDeviceID';
 
-  static Map<String, dynamic> generate(String appVersion, String deviceId) {
+  static Map<String, dynamic> generate(BuildInfoManager buildInfo) {
```
Comment (orestesgaolin) [inline lib/src/common/utils/device_info.dart:11]: I'm not the greatest fan of passing the BuildInfoManager here, but at the same time I didn't want to pass separate strings :/

File: lib/src/wiredash_controller.dart [hunk 2]
```diff
@@ -18,16 +18,40 @@ class WiredashController {
...
+    if (appVersion != null) {
+      setBuildProperties(buildVersion: appVersion);
+    }
+    if (userId != null) {
+      setUserProperties(userId: userId);
+    }
+    if (userEmail != null) {
+      setUserProperties(userEmail: userEmail);
+    }
+  }
```
Comment (orestesgaolin) [inline lib/src/wiredash_controller.dart:33]: Deprecating for now, will be removed in future releases.

File: lib/src/common/utils/device_info.dart (hunk 1 above)
Comment (flschweiger) [reply lib/src/common/utils/device_info.dart:11]: I think that is fine for now, in an upcoming release we'll have to restructure a few of these things either way 😉

File: example/lib/main.dart [hunk 3]
```diff
@@ -30,7 +30,7 @@ class _ExampleAppState extends State<ExampleApp> {
       options: WiredashOptionsData(
         showDebugFloatingEntryPoint: true,
       ),
-      theme: WiredashThemeData(brightness: _brightness),
+      theme: WiredashThemeData(brightness: Theme.of(context).brightness),
```
Comment (flschweiger) [inline example/lib/main.dart:33]: The _brightness variable was used to switch between dark and light UI for demonstration purposes - so I think we can get rid of that in the future 😉

**Pull request merged**